Source/Laser/LaserEnvelope.H

```diff
@@ -0,0 +1,29 @@
+#include <complex>
+#include <iostream>
+#include <complex>
+#include <vector>
+
+#include <AMReX.H>
+#include <iostream>
+#include <vector>
+#include <AMReX_REAL.H>
+#include <Utils/WarpX_Complex.H>
+#include <AMReX_AmrCore.H>
+
+class LaserEnvelope {
+public:
+    LaserEnvelope(amrex::AmrCore* amr_core, int ispecies, const std::string& name);
+    amrex::Vector< amrex::FabArray< amrex::BaseFab <Complex> > > A_laser_envelope;
```

Comment:
We probably want to start with
```cpp
amrex::Vector<std::unique_ptr<amrex::MultiFab>> A_laser_envelope;
```
and then allocate 2 components in the fourth index when we allocate the data
```cpp
AllocInitMultiFab(A_laser_envelope[lev], amrex::convert(ba, A_nodal_flag), dm, ncomps, ngA, tag("A_laser_envelope"), 0.0_rt);
```
using `ncomps=2` (meaning index `0` will be the real part and index `1` will be the imaginary part) and a flag for the index type, `A_nodal_flag`, that is identical to the one used for the `rho` MultiFabs (typically called `rho_nodal_flag`). The number of guard cells `ngA` would need to be set somewhere in `GuardCellManager.cpp` similarly to the other `ng` variables in WarpX.cpp.